distributed/scheduler.py

```diff
@@ -5059,7 +5059,7 @@ def validate_waiting(self, key):
         assert ts not in parent._unrunnable
         for dts in ts._dependencies:
             # We are waiting on a dependency iff it's not stored
-            assert (not not dts._who_has) != (dts in ts._waiting_on)
```

Comment:
this is not super performance critical since it is running in a validate which is off by default and only enabled in test. ns runtime in tests *must* not be of relevance.